### PR TITLE
Namespace amp-social-share buttons classnames

### LINKED ISSUE #2668: Namespace classes in amp-social-share elements
Right now elements created with `amp-social-share` have class names like this:

```html
<span class="twitter">
```

They should probably be something like:

```html
<span class="amp-social-share-twitter">
```

so as not to conflict with class names used elsewhere on a site's stylesheets.

### PR DESCRIPTION
**Pull request by mkhatib**
Fixes #2668

### CODE REVIEW

File: extensions/amp-social-share/0.1/amp-social-share.css [hunk 1]
```diff
@@ -39,31 +39,31 @@ amp-social-share a {
 }
 
 /* Twitter Styling */
-amp-social-share .twitter {
+amp-social-share .amp-social-share-twitter {
```
Comment (sriramkrish85) [inline extensions/amp-social-share/0.1/amp-social-share.css]: just .amp-social-share-twitter should be enough now that you have namespaced

File: extensions/amp-social-share/0.1/amp-social-share.js [hunk 2]
```diff
@@ -93,7 +93,7 @@ class AmpSocialShare extends AMP.BaseElement {
       let container = elementByTag(this.element, 'span');
       if (container == null) {
         container = this.getWin().document.createElement('span');
-        container.classList.add(this.type_);
+        container.classList.add('amp-social-share-' + this.type_);
```
Comment (sriramkrish85) [inline extensions/amp-social-share/0.1/amp-social-share.js]: make this a constant - 'amp-social-share-'

Comment (sriramkrish85) [inline extensions/amp-social-share/0.1/amp-social-share.js]: add documentation in the md file to reflect what the naming constraints are - and how ppl are expected to style them.

File: extensions/amp-social-share/0.1/test/test-amp-social-share.js [hunk 3]
```diff
@@ -109,6 +109,7 @@ describe('amp-social-share', () => {
       const tShare = ins.getElementsByTagName('span')[0];
       expect(tShare).to.not.be.null;
       expect(tShare.firstChild).to.not.be.null;
+      expect(tShare.className).to.contain('amp-social-share-twitter');
```
Comment (sriramkrish85) [inline extensions/amp-social-share/0.1/test/test-amp-social-share.js]: you will nee to add tests to make sure that the class names are properly namespaced?

Comment (mkhatib) [inline extensions/amp-social-share/0.1/test/test-amp-social-share.js]: Not entirely sure what do you mean here

File: extensions/amp-social-share/0.1/amp-social-share.js (hunk 2 above)
Comment (mkhatib) [inline extensions/amp-social-share/0.1/amp-social-share.js]: Done.

File: extensions/amp-social-share/0.1/amp-social-share.css (hunk 1 above)
Comment (mkhatib) [inline extensions/amp-social-share/0.1/amp-social-share.css]: Done.

File: extensions/amp-social-share/0.1/test/test-amp-social-share.js (hunk 3 above)
Comment (mkhatib) [inline extensions/amp-social-share/0.1/test/test-amp-social-share.js]: Done.

**Comment by mkhatib** [comment]
PTAL

File: extensions/amp-social-share/0.1/test/test-amp-social-share.js [hunk 4]
```diff
@@ -144,6 +145,7 @@ describe('amp-social-share', () => {
...
```
Comment (erwinmombay) [inline extensions/amp-social-share/0.1/test/test-amp-social-share.js]: use `expect(tShare).to.have.class('amp-social-share-twitter')` instead

Comment (mkhatib) [inline extensions/amp-social-share/0.1/test/test-amp-social-share.js]: Done. Thanks

**Comment by mkhatib** [comment]
PTAL

File: extensions/amp-social-share/0.1/test/test-amp-social-share.js [hunk 5]
```diff
@@ -154,6 +156,31 @@ describe('amp-social-share', () => {
     });
   });
 
+  it('does not add our own classnames when <a> element is provided', () => {
```
Comment (sriramkrish85) [inline extensions/amp-social-share/0.1/test/test-amp-social-share.js]: Should not add prefixed class for social shares with <a> tag

File: extensions/amp-social-share/amp-social-share.md [hunk 6]
```diff
@@ -78,7 +78,7 @@ You can include any additional document structure around the anchor, so long as
 
 Required attributes are `type`, `width` and `height`. Some [types (social providers)](#user-content-types) require specific fields for their integration. For instance Facebook requires you include your ```app_id``` (as ```attribution```), failure to this attribute for ```type="facebook"``` will result in an error.
 
-You can embed an `anchor` tag _without a_ ```href``` into the element to for the extension to provide the href for you. This enables customization of the social share element. You can also specify an arbitrary amount of AMP compatible HTML within the element to provide any hooks for styling.
+You can embed an `anchor` tag _without a_ ```href``` into the element to for the extension to provide the href for you with classname `.amp-social-share-TYPE`, where `TYPE` is the value of the type attribute provided, for example, `.amp-social-share-twitter`. This enables customization of the social share element. You can also specify an arbitrary amount of AMP compatible HTML within the element to provide any hooks for styling.
```
Comment (sriramkrish85) [inline extensions/amp-social-share/amp-social-share.md]: this does not make any sense.

```
You can embed an `anchor` tag _without a_ ```href``` into the element to for the extension to provide the href for you. This enables customization of the social share element. AMP adds a class name `amp-social-share-<type>` to the extension, where <type> is the value provided in the `type` attribute. For example the social-share extension for twitter would have the classname `amp-social-share-twitter`. This class could be used as a hook for styling using CSS.
Also an arbitrary amount of AMP compatible HTML can be added within the element to provide any hooks for styling.
```

Comment (mkhatib) [inline extensions/amp-social-share/amp-social-share.md]: done.

**Comment by sriramkrish85** [comment]
@dvoytenko do you wanna take a quick look?

File: extensions/amp-social-share/0.1/test/test-amp-social-share.js (hunk 3 above)
Comment (erwinmombay) [inline extensions/amp-social-share/0.1/test/test-amp-social-share.js]: missed one

File: extensions/amp-social-share/amp-social-share.md [hunk 7]
```diff
@@ -78,7 +78,10 @@ You can include any additional document structure around the anchor, so long as
...
 
-You can embed an `anchor` tag _without a_ ```href``` into the element to for the extension to provide the href for you. This enables customization of the social share element. You can also specify an arbitrary amount of AMP compatible HTML within the element to provide any hooks for styling.
+AMP adds a class name `amp-social-share-<type>` to the extension, where `<type>` is the value provided in the `type` attribute. For example the social-share extension for twitter would have the classname `amp-social-share-twitter`. This class could be used as a hook for styling using CSS. 
+
+You can embed an `anchor` tag _without a_ ```href``` into the element for the extension to provide the href for you. This enables customization of the social share element.
+Also an arbitrary amount of AMP compatible HTML can be added within the element to provide any hooks for styling. AMP won't add `amp-social-share-<type>` classname when anchor element is provided.
```
Comment (dvoytenko) [inline extensions/amp-social-share/amp-social-share.md]: Has this always been the behavior? Seems a bit misleading on the first read.

Comment (mkhatib) [inline extensions/amp-social-share/amp-social-share.md]: yeah this is a bit confusing but yes it seems to have been the case from the start.

**Comment by mkhatib** [comment]
@dvoytenko did you want me to hold onto this until next Wednesday, or is this ok to go in?

**Comment by dvoytenko** [comment]
LGTM. Please prepare the format review doc, however.

**Pull request merged**